### ISSUE TITLE
SceneTreeTweens leak memory in 3.5

### ISSUE THREAD

**Issue by SanderVanhove** [godotengine/godot#70636]
### Godot version

v3.5.1.stable.official [6fed1ffa3]

### System information

Windows 10

### Issue description

SceneTreeTweens and all its surrounding classes, like PropertyTweener and such, don't get removed, even when the tween gets bound to the current node.

I saw multiple issues that seem to be fixed and closed for Godot 4 (#61340, #51231, ...), but none that reports this in Godot 3.5.1 using GDScript.

### Steps to reproduce

The smallest piece of code that reproduces this:

```
extends Node2D


func _input(event: InputEvent) -> void:
	if not event is InputEventMouseButton: return

	create_tween().bind_node(self).tween_property(self, "position:x", 0.0, 5)
```

So, every time you click, SceneTreeTweens get spawned.

### Minimal reproduction project

[scene_tree_tween_leak.zip](https://github.com/godotengine/godot/files/10309809/scene_tree_tween_leak.zip)

**Comment by timothyqiu** [comment]
Fixed by #70103

**Comment by SanderVanhove** [comment]
Oh, didn't see this PR, great to see it's coming to 3.5.2!
